easybuild/easyblocks/q/quantumespresso.py

```diff
@@ -61,132 +63,269 @@ def __init__(self, *args, **kwargs):
+                self._dflags += ["-D__ELPA"]
+            elif LooseVersion(elpa_v) >= LooseVersion("2015"):
+                elpa_year_v = elpa_v.split('.')[0]
+                self._dflags += [f"-D__ELPA_{elpa_year_v}"]
+        elif LooseVersion(self.version) >= LooseVersion("6.0"):
+            if LooseVersion(elpa_v) >= LooseVersion("2017"):
+                raise EasyBuildError("ELPA support for QuantumESPRESSO 6.x only available up to v2016.xx")
+            elif LooseVersion(elpa_v) >= LooseVersion("2016"):
+                self._dflags += ["-D__ELPA_2016"]
+            elif LooseVersion(elpa_v) >= LooseVersion("2015"):
+                self._dflags += ["-D__ELPA_2015"]
+        elif LooseVersion(self.version) >= LooseVersion("5.4"):
+            self._dflags += ["-D__ELPA"]
+            self.cfg.update('configopts', f'--with-elpa={elpa}')
+            return
+        elif LooseVersion(self.version) >= LooseVersion("5.1.1"):
+            self.cfg.update('configopts', f'--with-elpa={elpa}')
+            return
+        else:
+            raise EasyBuildError("ELPA support is only available in QuantumESPRESSO 5.1.1 and later")
+        
+        if self.toolchain.options.get('openmp', False):
+            elpa_include = f'elpa_openmp-{elpa_v}'
+            elpa_lib = 'libelpa_openmp.a'
+        else:
+            elpa_include = f'elpa-{elpa_v}'
+            elpa_lib = 'libelpa.a'
+        elpa_include = os.path.join(elpa, 'include', elpa_include, 'modules')
+        elpa_lib = os.path.join(elpa, 'lib', elpa_lib)
+        self._repls += [
+            ('IFLAGS', f'-I{elpa_include}', True)
+            ]
+        self.cfg.update('configopts', f'--with-elpa-include={elpa_include}')
+        self.cfg.update('configopts', f'--with-elpa-lib={elpa_lib}')
+        if LooseVersion(self.version) < LooseVersion("7.0"):
+            self._repls += [
+                ('SCALAPACK_LIBS', f'{elpa_lib} {os.getenv("LIBSCALAPACK")}', False)
+                ]
+
+    def _add_fftw(self, comp_fam):
+        """Add FFTW support to the build."""
+        if self.toolchain.options.get('openmp', False):
+            libfft = os.getenv('LIBFFT_MT')
+        else:
+            libfft = os.getenv('LIBFFT')
 
-                elpa_min_ver = "2016.11.001.pre"
-            else:
-                elpa_min_ver = "2015"
-                dflags.append('-D__ELPA_2015 -D__ELPA')
+        if LooseVersion(self.version) >= LooseVersion("5.2.1"):
+            if comp_fam == toolchain.INTELCOMP and get_software_root("imkl"):
+                self._dflags += ["-D__DFTI"]
+            elif libfft:
+                self._dflags += ["-D__FFTW"] if "fftw3" not in libfft else ["-D__FFTW3"]
+                env.setvar('FFTW_LIBS', libfft)
+        elif LooseVersion(self.version) >= LooseVersion("5.0"):
+            if libfft:
+                self._dflags += ["-D__FFTW"] if "fftw3" not in libfft else ["-D__FFTW3"]
+                env.setvar('FFTW_LIBS', libfft)
```

Comment:
`self._repls`?

Reply:
Here i was using the env var to let QE automatically get the correct value during the configure, but it makes change to manually override the `make.inc` like we are doing for the other libraries

Reply:
Fixed in 96ededc